src/sentry/lang/native/symbolicator.py

```diff
@@ -290,9 +291,12 @@ def parse_sources(config):
 
 
 def get_options_for_project(project):
+    rate = options.get("symbolicator.compare_stackwalking_methods_rate")
+    compare_stackwalking = True if random.random() < rate else False
```

Comment:
Some weirdness here:

`rate=1.0` can randomly result in `False` because `1.0 < 1.0` is false

this is why I used to do `rate and random.random() < rate`, which is shitty but treats both `0.0` and `1.0` correctly

`True if <bool cond> else False` can just be `<bool cond>`

Reply:
According to the python docs the `random.random()` returns `[0.0, 1.0)` so it should never return 1.0 but only something very close to it.  So we think this should behave correctly, unless we misunderstood something.

> True if <bool cond> else False can just be <bool cond>
lol, of course

Reply:
>the random.random() returns [0.0, 1.0)

TIL, I should read docs more often